test/test_ops.py

```diff
@@ -194,10 +200,13 @@ def check_variant_backward(self, input, forward_result, expected_grad, expected_
     @ops(op_db)
     def test_variant_consistency_eager(self, device, dtype, op):
         test_backward = op.supports_autograd and op.test_complex_grad or not dtype.is_complex
-        samples = op.sample_inputs(device, dtype, requires_grad=test_backward)
-        if len(samples) == 0:
+        try:
+            samples = op.sample_inputs(device, dtype, requires_grad=test_backward)
+            next(iter(samples))
+        except StopIteration:
             self.skipTest("Skipped! No sample inputs!")
 
+        samples = op.sample_inputs(device, dtype, requires_grad=test_backward)
```

Comment:
We can reuse `samples` with above, but we will need to concatenate it with the the probe sample from the `try/except` block